xarray_schema/components.py

```diff
@@ -155,9 +155,8 @@ def validate(
                     # handles case of expected chunksize is shorthand of -1 which translates to the full length of dimension
                     if ec < 0:
                         ec = dim_sizes[key]
-                        # grab the first entry in da's tuple of chunks to be representative (since we've checked above that they're regular)
-                    ac = dim_chunks[key][0]
-                    if ac != ec:
+                    ac = dim_chunks[key]
+                    if not all(a == ec for a in ac[:-1]) and ac[-1] <= ec:
```

Comment:
i think the logic needs a fix. either the parenthetical should include the last part like 
`if not all(a == ec for a in ac[:-1] and ac[-1] <= ec):` or switch to `if not all(a == ec for a in ac[:-1]) or ac[-1] <= ec:`. Seems like you'd want to raise the error if either of those things occur, while here if `ac[-1]>ec` the first part failing won't trigger the error.